tanstack-table/src/main/scala/lucuma/react/table/HTMLTableRenderer.scala

```diff
@@ -209,21 +253,96 @@ object HTMLTableRenderer:
           estimateSize = props.estimateRowHeightPx,
           getScrollElement = ref.get,
           overscan = props.overscan,
-          getItemKey = props.getItemKey
+          getItemKey = props.getItemKey,
+          onChange = props.onChange
         )
       )
+      .useEffectOnMountBy((props, _, virtualizer) => // Allow external access to Virtualizer
+        props.virtualizerRef.toOption.map(_.set(virtualizer.some)).getOrEmpty
+      )
       .render { (props, ref, virtualizer) =>
         val rows                        = props.table.getRowModel().rows
         val (paddingTop, paddingBottom) = virtualVerticalPadding(virtualizer)
 
-        <.div.withRef(ref)(props.containerClass)(
+        // TODO Should we attempt to make the <table>  the container (scroll element) and <tbody> the virtualized element?
+        <.div.withRef(ref)(^.overflowY.auto, props.containerMod)(
           renderer.render(
             props.table,
             virtualizer.getVirtualItems().map(virtualItem => rows(virtualItem.index.toInt)),
-            props.tableClass,
-            props.rowClassFn,
+            TagMod(props.tableMod, props.extraTableClasses),
+            props.headerMod,
+            props.headerRowMod,
+            props.headerCellMod,
+            props.bodyMod,
+            props.rowMod,
+            props.cellMod,
+            props.footerMod,
+            props.footerRowMod,
+            props.footerCellMod,
             paddingTop.some,
-            paddingBottom.some
+            paddingBottom.some,
+            props.emptyMessage
+          )
+        )
+      }
+
+  def componentBuilderAutoHeightVirtualized[T, Props <: HTMLAutoHeightVirtualizedTableProps[_]](
+    renderer: HTMLTableRenderer[T]
+  ) =
+    ScalaFnComponent
+      .withHooks[Props[T]]
+      .useRefToVdom[HTMLDivElement]
+      .useVirtualizerBy((props, ref) =>
+        VirtualOptions(
+          count = props.table.getRowModel().rows.length,
+          estimateSize = props.estimateRowHeightPx,
+          getScrollElement = ref.get,
+          overscan = props.overscan,
+          getItemKey = props.getItemKey,
+          onChange = props.onChange
+        )
+      )
+      .useEffectOnMountBy((props, _, virtualizer) => // Allow external access to Virtualizer
+        props.virtualizerRef.toOption.map(_.set(virtualizer.some)).getOrEmpty
+      )
+      .render { (props, ref, virtualizer) =>
+        val rows                        = props.table.getRowModel().rows
+        val (paddingTop, paddingBottom) = virtualVerticalPadding(virtualizer)
+
+        // We use this trick to get a component whose height adjusts to the container.
+        // See https://stackoverflow.com/a/1230666
+        // We create 2 more containers: an outer one, with position: relative and height: 100%,
+        // and an inner one, with position: absolute, and top: 0, bottom: 0.
+        // The scrolling element has to be the outer one.
```

Comment:
Neat. Could this be done with the `flex` solution instead?

Reply:
I couldn't make it work with flex. We can change it if there's a better combination of elements/styles.